torch_ttnn/passes/lowering/to_tt_pass.py

```diff
@@ -934,6 +933,82 @@ def rewrite_node(node):
                 # TODO(tt-metal#12099): Currently it doesn't return indices. Pack into tuple to maintain the type
                 return g.call_function(target_wrappers.pack_to_tuple, (output_tensor,))
 
+            if node.target == torch.ops.aten.convolution.default:
+                params = map_args_to_kwargs(
+                    args,
+                    (
+                        (0, "input_tensor"),
+                        (1, "weight_tensor"),
+                        (2, "bias_tensor"),
+                        (3, "stride"),
+                        (4, "padding"),
+                        (5, "dilation"),
+                        (6, "transposed"),
+                        (7, "output_padding"),
+                        (8, "groups"),
+                    ),
+                )
+                input_node = params["input_tensor"]
+                weight_node = params["weight_tensor"]
+                transposed = params.get("transposed", False)
+                groups = params.get("groups", 1)
+                stride = params.get("stride", [1, 1])
+                padding = params.get("padding", [0, 0])
+                dilation = params.get("dilation", [1, 1])
+                output_padding = params.get("output_padding", [0, 0])
+
+                input_shape = input_node.meta["val"].size()
+                if len(input_shape) < 4:
+                    return None
+
+                batch_size, in_c, in_h, in_w = input_shape
+                out_a, out_b, kernel_h, kernel_w = weight_node.meta["val"].size()
+                out_c = out_b if transposed else out_a
+
+                # TODO(#416): Weight tensor on owned storage isn't supported
+                if groups > 1:
+                    return None
+
+                input_tensor = insert_nchw_to_nhwc(g, input_node)
+                # TODO(tt-metal#15148): ttnn.conv2d internal reshape fails with padding
+                input_tensor = g.call_function(ttnn.reshape, (input_tensor, (1, 1, batch_size * in_h * in_w, in_c)))
+                weight_tensor = g.call_function(
+                    target_wrappers.move_to_host,
+                    (weight_node, TtnnRowMajorLayout()),
+                )
```

Comment:
two things:
* ideally we don't put it to device in the first place. would be nice if there is a pass which can eliminate this to_device, from_device pair
* ideally we store and reuse preprocessed weights tensor produced by conv (there is a PR in metal which moves preprocessing into a separate OP to give more control over it

Nothing to be done in this PR, but want to share this notes as opportunity for further joint improvement.

Reply:
Agree. `torch_ttnn/passes/lowering/eliminate_coreops_pass.py` would be potential pass to eliminate `to_device` and `move_to_host` when possible.

Have an ttnn op to allow pre-process weight separately will definitely help to cache and reuse pre-processed weight tensor